x-pack/test/api_integration/apis/cloud_security_posture/benchmark.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import type { BenchmarkResponse } from '@kbn/cloud-security-posture-plugin/common/types';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { createPackagePolicy } from './status';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+  const kibanaServer = getService('kibanaServer');
+
+  describe('GET /internal/cloud_security_posture/status', () => {
+    let agentPolicyId: string;
+
+    beforeEach(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await esArchiver.load('x-pack/test/functional/es_archives/fleet/empty_fleet_server');
+
+      const { body: agentPolicyResponse } = await supertest
+        .post(`/api/fleet/agent_policies`)
+        .set('kbn-xsrf', 'xxxx')
+        .send({
+          name: 'Test policy',
+          namespace: 'default',
+        });
+
+      agentPolicyId = agentPolicyResponse.item.id;
+    });
+
+    afterEach(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await esArchiver.unload('x-pack/test/functional/es_archives/fleet/empty_fleet_server');
+    });
+
+    it(`Should return non-empty array filled with Rules if user has CSP integrations`, async () => {
```

Comment:
let's add few more test cases checking the pagination works well

For example:
A test case that creates 3 integrations, 
Then query `/benchmarks` api with page size of: `1` or `2`
And testing the following cases:
- Receiving the expected results on the page
- Receiving a the last page with partial results / full results
- Asking for page that has no results (page `4` or `3` depending on the page size)

Since we have our own math to do the page breaks, I would prefer having automated tests for that